packages/mdc-data-table/README.md

```diff
@@ -243,7 +243,8 @@ Method Signature | Description
 
 Method Signature | Description
 --- | ---
-`layout() => void` | Re-initializes header row checkbox and row checkboxes when selectable rows are added or removed from table.
+`layout() => void` | Re-initializes header row checkbox and row checkboxes when selectable rows are added or removed from table. Use this if registering checkbox is synchronous.
+`layoutAsync() => void` | Re-initializes header row checkbox and row checkboxes when selectable rows are added or removed from table. Use this if registering checkbox is asynchronous.
```

Comment:
does Async mean checkbox is added after the fact? Or do you mean if checkbox is initialized after the datatable after an event has occurred? Just want to know of the use case.

Reply:
layoutAsync() itself is asynchronous because registerHeaderRowCheckbox and registerRowCheckboxes can be asynchronous in this case.

This supports use case where registerHeaderRowCheckbox / registerRowCheckboxes does not register checkbox instances synchronously. So the client would've to wait till all promises are resolved.

Example,

```ts
async updateTable() {
  await layout();
}
```

Reply:
Document that `layoutAsync` returns a Promise, not void. Can you also modify the description to specify the registerCheckbox methods which are sync/async?

Reply:
Done.